flocker/node/agents/gce.py

```diff
@@ -544,3 +549,42 @@ def destroy_volume(self, blockdevice_id):
             else:
                 raise e
         return None
+
+    def list_live_nodes(self):
+        page_token = None
+        done = False
+        nodes = []
+        while not done:
+            result = self._compute.instances().list(
+                project=self._project,
+                zone=self._zone,
+                maxResults=self._page_size,
+                pageToken=page_token
+            ).execute()
+            page_token = result.get('nextPageToken')
+            nodes.extend(result.get('items', []))
+            done = not page_token
+        return set(node.get("name") for node in nodes
+                   if node.get("status") == "RUNNING")
+
+    def start_node(self, node_id):
+        self._do_blocking_operation(
+            self._compute.instances().start,
+            timeout_sec=5*60,
+            instance=node_id
+        )
+
+    def stop_node(self, node_id):
+        """
+        Stops a node. This shuts the node down, but leaves the boot disk
+        available so that it can be started again using ``start_node``.
+
+        Note this is only used in the functional tests of start_node.
+
+        :param unicode node_id: The compute_instance_id of the node to stop.
+        """
+        self._do_blocking_operation(
+            self._compute.instances().stop,
+            timeout_sec=5*60,
+            instance=node_id
+        )
```

Comment:
+1 I also winder if this should actually be part of the provisioner instead of part of this interface.

For now though, just doing a GCE-specific implementation.

Changing to private though, as it is only used by this and the immediate functional tests of this.